File: 1-1000/0957-prison-cells-after-n-days/0957-prison-cells-after-n-days.cpp
class Solution {
public:
    vector<int> prisonAfterNDays(vector<int>& cells, int n) {
        int size = cells.size();
        vector<int> curr (size, 0);
        vector<int> prev = cells;
        n = (n-1)%14 + 1;
        while(n>0)
        {  
            for(int i = 0; i<size; i++){
                if(i-1 < 0 || i+1 == size) curr[i] = 0;
                else if(prev[i-1] == prev[i+1]) curr[i] = 1;
                else curr[i] = 0;
            }
            prev = curr;
            n--;
        }
        return curr;
    }
};

File: 1-1000/0957-prison-cells-after-n-days/0957-prison-cells-after-n-days.java
class Solution {
    public int[] prisonAfterNDays(int[] cells, int n) {
        n = n % 14;
        if (n == 0) n = 14;

        for (int day = 0; day < n; day++) {
            int[] next = new int[8];
            for (int i = 1; i < 7; i++) {
                next[i] = (cells[i - 1] == cells[i + 1]) ? 1 : 0;
            }
            cells = next;
        }
        return cells;
    }
}